CMakeLists.txt

```diff
@@ -344,6 +345,10 @@ target_link_libraries(ngen
         NGen::logging
 )
 
+if(NGEN_WITH_MPI)
+    target_link_libraries(ngen PUBLIC NGen::MPI_Detour)
```

Comment:
Just realized, but this needs to be linked before MPI::MPI_C, otherwise it does not overwrite the MPI implementation of `MPI_Finalize`.